src/paginate.ts

```diff
@@ -385,25 +390,53 @@ export async function paginate<T extends ObjectLiteral>(
 
     const buildLink = (p: number): string => path + '?page=' + p + options
 
-    const results: Paginated<T> = {
-        data: items,
-        meta: {
-            itemsPerPage: limit,
-            totalItems,
-            currentPage: page,
-            totalPages: totalPages,
-            sortBy,
-            search: query.search,
-            searchBy: query.search ? searchBy : undefined,
-            filter: query.filter,
-        },
-        links: {
-            first: page == 1 ? undefined : buildLink(1),
-            previous: page - 1 < 1 ? undefined : buildLink(page - 1),
-            current: buildLink(page),
-            next: page + 1 > totalPages ? undefined : buildLink(page + 1),
-            last: page == totalPages || !totalItems ? undefined : buildLink(totalPages),
-        },
+    let results: Paginated<T>
+
+    if (limit) {
+        let totalPages = totalItems / limit
+        if (totalItems % limit) totalPages = Math.ceil(totalPages)
+
+        results = {
+            data: items,
+            meta: {
+                itemsPerPage: limit,
+                totalItems,
+                currentPage: page,
+                totalPages: totalPages,
+                sortBy,
+                search: query.search,
+                searchBy: query.search ? searchBy : undefined,
+                filter: query.filter,
+            },
+            links: {
+                first: page == 1 ? undefined : buildLink(1),
+                previous: page - 1 < 1 ? undefined : buildLink(page - 1),
+                current: buildLink(page),
+                next: page + 1 > totalPages ? undefined : buildLink(page + 1),
+                last: page == totalPages || !totalItems ? undefined : buildLink(totalPages),
+            },
+        }
+    } else {
+        results = {
+            data: items,
+            meta: {
```

Comment:
let's just branch the calculation of the meta values which are computed differently now, instead of this larger code duplication